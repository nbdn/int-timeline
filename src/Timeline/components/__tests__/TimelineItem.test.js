import React from 'react';
import { mount, shallow } from 'enzyme';
import Item from '../TimelineItem';

import { timelineDataMock, timelineRuleDataMock } from '../../__tests__/mocks';

describe('component rendering right', () => {
  let wrapper;
  let emptyPropsWrapper;
  beforeEach(() => {
    wrapper = shallow(
      <Item
        groupStart={timelineDataMock[0].start}
        iItem={0}
        start={timelineDataMock[0].items[0][0].start}
        end={timelineDataMock[0].items[0][0].end}
        title={timelineDataMock[0].items[0][0].title}
        unitWidth={timelineRuleDataMock.unitWidth}
      />
    );

    // Bypass random border color prop to match snapshot
    wrapper.childAt(0).prop('style').borderColor = '#612DC8';

    emptyPropsWrapper = shallow(<Item />);

    // Bypass random border color prop to match snapshot
    emptyPropsWrapper.childAt(0).prop('style').borderColor = '#612DC8';
  });

  it('renders props right match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('render with empty props match snapshot', () => {
    expect(emptyPropsWrapper).toMatchSnapshot();
  });
});

describe('component has right props', () => {
  it('has right default props', () => {
    const wrapper = mount(<Item />);
    expect(wrapper.prop('groupStart')).toEqual(0);
    expect(wrapper.prop('iItem')).toEqual(0);
    expect(wrapper.prop('start')).toEqual(0);
    expect(wrapper.prop('end')).toEqual(0);
    expect(wrapper.prop('title')).toEqual('');
    expect(wrapper.prop('unitWidth')).toEqual(0);
  });

  it('has right given props', () => {
    const wrapper = mount(
      <Item
        groupStart={timelineDataMock[0].start}
        iItem={0}
        start={timelineDataMock[0].items[0][0].start}
        end={timelineDataMock[0].items[0][0].end}
        title={timelineDataMock[0].items[0][0].title}
        unitWidth={timelineRuleDataMock.unitWidth}
      />
    );
    expect(wrapper.prop('groupStart')).toEqual(timelineDataMock[0].start);
    expect(wrapper.prop('iItem')).toEqual(0);
    expect(wrapper.prop('start')).toEqual(
      timelineDataMock[0].items[0][0].start
    );
    expect(wrapper.prop('end')).toEqual(timelineDataMock[0].items[0][0].end);
    expect(wrapper.prop('title')).toEqual(
      timelineDataMock[0].items[0][0].title
    );
    expect(wrapper.prop('unitWidth')).toEqual(timelineRuleDataMock.unitWidth);
  });
});

describe('component structure', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <Item
        groupStart={timelineDataMock[0].start}
        iItem={0}
        start={timelineDataMock[0].items[0][0].start}
        end={timelineDataMock[0].items[0][0].end}
        title={timelineDataMock[0].items[0][0].title}
        unitWidth={timelineRuleDataMock.unitWidth}
      />
    );
  });

  it('has right wrapper class name', () => {
    expect(wrapper.hasClass('TimelineItem')).toBeTruthy();
  });

  it('has right children className', () => {
    expect(wrapper.childAt(0).hasClass('TimelineItem__content')).toBeTruthy();
  });

  it('has right children length', () => {
    expect(wrapper.children()).toHaveLength(1);
  });

  it('children display title', () => {
    const titleMock = <span>{timelineDataMock[0].items[0][0].title}</span>;
    expect(wrapper.childAt(0).contains(titleMock)).toBeTruthy();
  });

  it('children has right transform style property', () => {
    const translateXMock = `translateX(${timelineDataMock[0].start *
      timelineRuleDataMock.unitWidth -
      timelineDataMock[0].start * timelineRuleDataMock.unitWidth}px)`;
    expect(wrapper.prop('style').transform).toEqual(translateXMock);
  });
});
