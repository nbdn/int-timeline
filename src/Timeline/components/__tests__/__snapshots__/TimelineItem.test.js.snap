// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component rendering right render with empty props match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TimelineItem
    end={0}
    groupStart={0}
    iItem={0}
    start={0}
    title=""
    unitWidth={0}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": "timeline_item_0",
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="TimelineItem__content"
        style={
                Object {
                        "borderColor": "#612DC8",
                      }
        }
>
        <span />
</div>,
      "className": "TimelineItem",
      "style": Object {
        "transform": "translateX(0px)",
        "width": 0,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <span />,
        "className": "TimelineItem__content",
        "style": Object {
          "borderColor": "#612DC8",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
        },
        "ref": null,
        "rendered": "",
        "type": "span",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": "timeline_item_0",
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="TimelineItem__content"
          style={
                    Object {
                              "borderColor": "#612DC8",
                            }
          }
>
          <span />
</div>,
        "className": "TimelineItem",
        "style": Object {
          "transform": "translateX(0px)",
          "width": 0,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <span />,
          "className": "TimelineItem__content",
          "style": Object {
            "borderColor": "#612DC8",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
          },
          "ref": null,
          "rendered": "",
          "type": "span",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`component rendering right renders props right match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TimelineItem
    end={50}
    groupStart={5}
    iItem={0}
    start={5}
    title="Study of Bendamustine"
    unitWidth={150}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": "timeline_item_0",
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="TimelineItem__content"
        style={
                Object {
                        "borderColor": "#612DC8",
                      }
        }
>
        <span>
                Study of Bendamustine
        </span>
</div>,
      "className": "TimelineItem",
      "style": Object {
        "transform": "translateX(0px)",
        "width": 6750,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <span>
          Study of Bendamustine
</span>,
        "className": "TimelineItem__content",
        "style": Object {
          "borderColor": "#612DC8",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Study of Bendamustine",
        },
        "ref": null,
        "rendered": "Study of Bendamustine",
        "type": "span",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": "timeline_item_0",
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="TimelineItem__content"
          style={
                    Object {
                              "borderColor": "#612DC8",
                            }
          }
>
          <span>
                    Study of Bendamustine
          </span>
</div>,
        "className": "TimelineItem",
        "style": Object {
          "transform": "translateX(0px)",
          "width": 6750,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <span>
            Study of Bendamustine
</span>,
          "className": "TimelineItem__content",
          "style": Object {
            "borderColor": "#612DC8",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Study of Bendamustine",
          },
          "ref": null,
          "rendered": "Study of Bendamustine",
          "type": "span",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
